core/src/saros/communication/InfoManager.java

```diff
@@ -0,0 +1,140 @@
+package saros.communication;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import org.apache.log4j.Logger;
+import org.jivesoftware.smack.PacketListener;
+import org.jivesoftware.smack.filter.AndFilter;
+import org.jivesoftware.smack.packet.IQ;
+import org.jivesoftware.smack.packet.Packet;
+import saros.communication.extensions.InfoExchangeExtension;
+import saros.net.IReceiver;
+import saros.net.ITransmitter;
+import saros.net.xmpp.JID;
+import saros.net.xmpp.contact.IContactsUpdate;
+import saros.net.xmpp.contact.IContactsUpdate.UpdateType;
+import saros.net.xmpp.contact.XMPPContact;
+import saros.net.xmpp.contact.XMPPContactsService;
+
+/**
+ * This Component exchanges information in a key-value form with other contacts outside of Sessions.
+ *
+ * <p>Every info set via {@link #setLocalInfo(String, String)} is send / updated to available
+ * contacts with Saros support. Retrieval of available data with {@link #getRemoteInfo(XMPPContact,
+ * String)}.
+ */
+public class InfoManager {
+  private class ClientInfo {
+    private final Map<String, String> infos;
```

Comment:
I removed in the mean time because it has not a big value and I use ClientInfo mainly to avoid `Map<XMPPContact, Map<String, String>>` but yes probably better to keep it.